ESLint

brettz9 opened the pull request:
- Linting: Add `browser` and `node_modules` to `.eslintignore`
- npm: Simplify `test` script

whitlockjc commented:
I'm already running `eslint` as part of `gulp`.  Is this necessary?

brettz9 commented:
Whoops... Sorry about that...

However, the PR should still be useful as `.eslintignore` is useful to IDE linting plugins to avoid triggering errors and warnings when browsing a file in the repo that is not supposed to have the linting rules applied.

I also simplified the `package.json` `test` script which doesn't need to hard-code the path to the gulp binary given that dependencies are added to the path (see https://docs.npmjs.com/misc/scripts#path ).

whitlockjc commented:
I do like the idea of having an `.eslintignore` file.  Do you mind reverting your change to `package.json` and adding `bower_components` to the list of files in `.eslintignore`, and `coverage`?  Actually, `.gitignore` is a good place to look just in case.

brettz9 commented:
I've added `bower_components` and `coverage`. Two other directories are `.idea/` and `vscode/` but I didn't know if they might contain JavaScript files?

As far as the `package.json` change, I removed the `eslint` script, but I added a simplification that works because of how npm adds binary files to the path. I can't see any reason not to add that...

brettz9 commented:
Also a gentle reminder--bower has been deprecated...

whitlockjc commented:
I've not kept up with bower.  What's its replacement?

brettz9 commented:
npm/yarn... See https://github.com/whitlockjc/json-refs/issues/117#issuecomment-303543284 . npm has made some improvements since yarn started to get traction back when bower recommended yarn e.g., https://blog.risingstack.com/yarn-vs-npm-node-js-package-managers/ .

whitlockjc commented:
I'll give it a peek.  :)

brettz9 commented:
Without needing to get into a decision about adding yarn support, I think at least going with including the files in npm would be a clear and simple option. People are using npm for a lot more than just Node these days.

Since one can sometimes forget to update both `yarn.lock` and `package-lock.json` (the file that gets created now by `npm` when doing a local `npm install` and which specifies the exact versions in use), I'd personally go with just the latter if not supporting both.
